utils/update_arch_timings.py

```diff
@@ -0,0 +1,118 @@
+import lxml.etree as ET
+import argparse
+from sdf_timing import sdfparse
+from sdf_timing.utils import get_scale_seconds
+import re
+import os
+
+
+def remove_site_number(site):
+    number = re.search(r'\d+$', site)
+    if number is not None:
+        site = site[:-len(str(number.group()))]
+    return site
+
+
+def get_cell_type(bel, site):
+    if bel.endswith("5LUT") and site == "SLICEL":
+        return "LUT5"
+    if bel.endswith("5LUT") and site == "SLICEM":
+        return "LUT_OR_MEM5LRAM"
+
+    return None
+
+
+def find_timings(timings, bel, site):
+
+    separator = "/"
+    celltype = get_cell_type(bel, site)
+    if celltype is None:
+        return None
+
+    cells = timings['cells'][celltype]
+    cellsite = None
+    for cell in cells:
+        if (site in cell.split(separator)) and (bel in cell.split(separator)):
+            cellsite = cell
+            break
+        if site == cell:
+            cellsite = site
+            # do not break here as we may still find more precise
+            # site/bel location
+    if cellsite is None:
+        return None
+
+    bel_timings = dict()
+    for delay in cells[cellsite]:
+        if not cells[cellsite][delay]['is_absolute']:
+            continue
+        entry = cells[cellsite][delay]['delay_paths']['slow']['max']
+        bel_timings[delay] = float(entry) * get_scale_seconds('1 ns')
+
+    return bel_timings
+
+
+def get_pb_type_chain(node):
```

Comment:
Should this be part of `lib/pb_type.py`?

Reply:
done. I've moved the function to lib/pb_type